### DISCUSSION TITLE
Why doesn't nequip use PyTorch Geometric?

### DISCUSSION THREAD

**Discussion by ryandeng1** [mir-group/nequip#208] [Q&A] [answered]
Pytorch Geometric has many optimizations under the hood for GNNs, why doesn't Nequip choose to use that framework?

**Comment by Linux-cpp-lisp** [comment, marked as answer]
@ryandeng1 we do in fact use a limited subset of PyTorch Geometric including their graph, data, and dataloader formats. However the full flexibility and weight of a generic graph learning framework is a significant and unnecessary dependency that NequIP has no use for; our graphs are a very specific and simple sub-case.

Is there a specific optimization you are thinking of, besides TorchScript compilation, which we support in full?

**Comment by ryandeng1** [comment]
I have yet to test any of this, but one example is more efficient message-passing based aggregations with sparse adjacency matrices, which can save on memory.

**Comment by Linux-cpp-lisp** [comment]
We already don't use adjacency matrices... we use just lists of edges, which is the same as sparse coordinate format adjacency matrices. The reason I'm asking is that as far as I know, we've already basically implemented every performance technique from PyTorch Geometric that is applicable to our problem... if there are ones we haven't, I'd be curious to know what they are to consider trying them.
